lib/charms/flume.py

```diff
@@ -1,143 +0,0 @@
-import os
-import re
-import signal
-from subprocess import Popen, check_output
-
-import jujuresources
-
-from jujubigdata import utils
-from charmhelpers.core import unitdata, templating, hookenv
-
-class Flume(object):
-    """
-    This class manages the deployment steps of Flume agent.
-    
-    :param DistConfig dist_config: The configuration container object needed.
-    """
-    def __init__(self, dist_config):
-        self.dist_config = dist_config
-        self.resources = {
-            'flume': 'flume-%s' % utils.cpu_arch(),
-            'zookeeper': 'zookeeper-%s' % utils.cpu_arch(),
-        }
-        self.verify_resources = utils.verify_resources(*self.resources.values())
-
-    def is_installed(self):
-        return unitdata.kv().get('flume_kafka.installed')
-
-    def install(self, force=False):
-        '''
-        Create the users and directories. This method is to be called only once.
-        
-        :param bool force: Force the installation execution even if this is not the first installation attempt.
-        '''
-        if not force and self.is_installed():
-            return
-        jujuresources.install(self.resources['flume'],
-                              destination=self.dist_config.path('flume'),
-                              skip_top_level=True)
-        # FlumeSource needs the zookeeper jars
-        jujuresources.install(self.resources['zookeeper'],
-                              destination=self.dist_config.path('zookeeper'),
-                              skip_top_level=True)
-        self.dist_config.add_users()
-        self.dist_config.add_dirs()
-        self.dist_config.add_packages()
-        self.setup_flume_config()
-        unitdata.kv().set('flume_kafka.installed', True)
-
-
-    def setup_flume_config(self):
-        '''
-        copy the default configuration files to flume_conf property
-        defined in dist.yaml
-        '''
-        default_conf = self.dist_config.path('flume') / 'conf'
-        flume_conf = self.dist_config.path('flume_conf')
-        flume_conf.rmtree_p()
-        default_conf.copytree(flume_conf)
-        # Now remove the conf included in the tarball and symlink our real conf
-        default_conf.rmtree_p()
-        flume_conf.symlink(default_conf)
-
-        flume_env = self.dist_config.path('flume_conf') / 'flume-env.sh'
-        if not flume_env.exists():
-            (self.dist_config.path('flume_conf') / 'flume-env.sh.template').copy(flume_env)
-        utils.re_edit_in_place(flume_env, {
-            r'.*FLUME_CLASSPATH.*': 'FLUME_CLASSPATH={}/*'.format(self.dist_config.path('zookeeper')),
```

Comment:
Added this to the base in juju-solutions/layer-apache-flume-base#4 and the call to `configure_zookeeper` in this PR (that I missed before this comment)